src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -199,6 +199,30 @@
             bracket_pairs_set="angle_bracket_pairs",
         ),
     ),
+    StructTypeGrammar=Sequence(
+        "STRUCT",
+        Bracketed(
+            Delimited(  # Comma-separated list of field names/types
+                Sequence(
+                    OneOf(
+                        # ParameterNames can look like Datatypes so can't use
+                        # Optional=True here and instead do a OneOf in order
+                        # with DataType only first, followed by both.
+                        Ref("DatatypeSegment"),
+                        Sequence(
+                            Ref("ParameterNameSegment"),
+                            Ref("DatatypeSegment"),
+                        ),
+                    ),
+                    Ref("OptionsSegment", optional=True),
```

Comment:
The options are optional. 🤣 I haven't been drinking, although you'd be forgiven for wondering. 🍻